pkg/inject/virtualgateway_envoy.go

```diff
@@ -20,22 +21,26 @@ const envoyVirtualGatewayEnvMap = `
   "ENVOY_ADMIN_ACCESS_LOG_FILE": "{{ .AdminAccessLogFile }}",
   "AWS_REGION": "{{ .AWSRegion }}"{{ if .EnableSDS }},
   "APPMESH_SDS_SOCKET_PATH": "{{ .SdsUdsPath }}"{{ end }}{{ if .EnableXrayTracing }},
-  "ENABLE_ENVOY_XRAY_TRACING": "1","XRAY_DAEMON_PORT": "{{ .XrayDaemonPort }}"{{ end }}
+  "ENABLE_ENVOY_XRAY_TRACING": "1","XRAY_DAEMON_PORT": "{{ .XrayDaemonPort }}"{{ end }}{{ if .EnableJaegerTracing }},
+  "ENABLE_ENVOY_JAEGER_TRACING": "1","JAEGER_TRACER_ADDRESS": "{{ .JaegerAddress }}","JAEGER_TRACER_PORT": "{{ .JaegerPort }}"{{ end }}
```

Comment:
Just driving by this PR to say thanks for doing this!

I have a comment regarding this implementation vs the way we do it for virtual nodes. It seems like for virtual nodes we are generating a JSON config file with raw Envoy config for the tracer while for virtual gateways we simply pass the environment variables to the App Mesh envoy image.

Would it be very difficult to include making it consistent in this PR? Ideally we should just pass the `ENVOY_JAEGAR*` environment variables for both types of resources.

Coincidentally I was planning on making a PR to do this soon, hence why I'm asking now.

Reply:
Thanks for the suggestion. I noticed the existing implementation for x-ray, so followed the same pattern to keep my change minimum. I will take a look at the virtual node approach and accordingly update this PR.

Reply:
I was actually advocating to _just use the environment variables_ in the Envoy instead of mounting the envoyconf.yaml. I was planning to send a PR to remove this config mounting since the Envoy API can break between versions but the image can always generate the correct config for its given Envoy version.

Reply:
@lavignes Thanks for clarifying. In that case I will keep this change limited to virtual gateway envoy itself, will pass the environment variables instead of mounting the config file. 

I will have another PR soon to remove the mounting logic from virtual node as that would require testing as well.
Only for testing purpose and scope, i think it's better to have them separate.

Reply:
Ack. Sounds good to me